### PR TITLE
(SIMP-223) Refactor for SIMP 6

### PR DESCRIPTION
**Pull request by trevor-vaughan**
Refactor to support the new consolidated simp-rsync.

SIMP-223 #comment Update for SIMP 6

### CODE REVIEW

**Comment by trevor-vaughan** [comment]
This needs to use a fact instead of a function for the rsync space.

**Comment by trevor-vaughan** [comment]
Working on the update to the fact now.

File: build/rpm_metadata/requires [hunk 1]
```diff
@@ -15,19 +15,19 @@ Requires: pupmod-simp-auditd < 7.0.0-0
 Requires: pupmod-simp-auditd >= 6.0.0-0
 Requires: pupmod-simp-autofs < 6.0.0-0
 Requires: pupmod-simp-autofs >= 5.0.0-0
-Requires: pupmod-simp-clamav < 6.0.0-0
+Requires: pupmod-simp-clamav < 7.0.0-0
 Requires: pupmod-simp-clamav >= 5.0.0-0
```
Comment (lnemsick-simp) [inline build/rpm_metadata/requires]: Will this really interoperate with an older version of pupmod-simp-clamav, et. al?  I thought changes were required to add the environment paths to the rsync behavior in those modules?

File: lib/facter/simp_rsync_environments.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  _rsync_environments = File.join('', 'var', 'simp', 'rsync', 'environments')
+
+  confine do
+    File.directory?(_rsync_environments)
+  end
+
+  has_weight 1
+
+  setcode do
+    Dir.glob(File.join(_rsync_environments, '*')).map{|x| x = File.basename(x)}
```
Comment (trevor-vaughan) [inline lib/facter/simp_rsync_environments.rb]: This is OK. With password protected directories, you will want to have different passwords for each environment. The fact that it's a symlink should be irrelevant to the client systems.

File: manifests/server/rsync_shares.pp [hunk 3]
```diff
@@ -1,119 +1,128 @@
...
-    hosts_allow => $l_hosts_allow,
+    $_hosts_allow = $hosts_allow
   }
 
-  validate_absolute_path($rsync_base)
-  validate_bool($use_stunnel)
-  validate_net_list($hosts_allow)
+  if $facts['simp_rsync_environments'] and !empty($facts['simp_rsync_environments']) {
+    $facts['simp_rsync_environments'].each |String $_env| {
+
```
Comment (trevor-vaughan) [inline manifests/server/rsync_shares.pp]: @lnemsick-simp I guess that we could allow them to pass in an array. If the directories don't exist it'll be complete garbage though. I'll change this to give them the option.

File: build/rpm_metadata/requires (hunk 1 above)
Comment (trevor-vaughan) [inline build/rpm_metadata/requires]: @lnemsick-simp Hmm...yeah, I guess not at this point. Will fix.

**Comment by trevor-vaughan** [comment]
@lnemsick-simp Latest PR should resolve the review issues

File: manifests/kickstart_server.pp [hunk 4]
```diff
@@ -82,33 +82,27 @@
...
-  if !is_array($ntp_servers) { validate_hash($ntp_servers) }
-  else { validate_array($ntp_servers) }
   validate_net_list($puppet_server)
   validate_net_list($puppet_ca)
   validate_port($puppet_ca_port)
-  validate_bool($runpuppet_print_stats)
-  unless empty($runpuppet_wait_for_cert) { validate_integer($runpuppet_wait_for_cert) }
-  validate_string($sslverifyclient)
+
+  inspect($runpuppet_wait_for_cert)
```
Comment (jeefberkey) [inline manifests/kickstart_server.pp]: Whoops

File: manifests/puppetdb.pp [hunk 5]
```diff
@@ -29,37 +29,35 @@
...
-  Boolean $manage_puppetserver    = true,
-  String  $java_max_memory        = '40%',
-  String  $java_start_memory      = '',
-  String  $java_tmpdir            = '/opt/puppetlabs/puppet/cache/pdb_tmp',
-  Boolean $java_heapdump_on_oom   = false,
-  Boolean $java_prefer_ipv4       = true,
-  Boolean $use_iptables           = defined('$::use_iptables') ? { true  => $::use_iptables, default => hiera('use_iptables', true) }
+  Array[String]            $client_nets            = defined('$::client_nets') ? { true =>    $::client_nets,   default => hiera('client_nets',['127.0.0.1']) },
+  String                   $listen_address         = '127.0.0.1',
+  Stdlib::Compat::Integer  $listen_port            = '8138',
```
Comment (jeefberkey) [inline manifests/puppetdb.pp]: back into a integer string?

**Comment by trevor-vaughan** [comment]
@jeefberkey Fixed the random 'inspect'

**Comment by trevor-vaughan** [comment]
Fixing the spec tests, will push after that is done.

**Comment by trevor-vaughan** [comment]
Travis is failing because $::mco_version is not present for the mcollective module. However, this works fine on the systems where I tested it. Merging and we can figure this out later.

**Pull request merged**